prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -517,14 +518,26 @@ class ActionHistory(
         filename: String?,
         params: String,
         result: String,
-        itemCount: Int
+        header: WorkflowEngine.Header
     ) {
         if (isReportAlreadyTracked(sentReportId)) {
             error(
                 "Bug:  attempt to track history of a report ($sentReportId) " +
                     "we've already associated with this action"
             )
         }
+
+        if (header.content == null) {
+            error("Bug: attempt to track sent report with no contents")
+        }
+
+        val blobInfo = BlobAccess.uploadBody(
+            receiver.format,
+            header.content,
+            filename ?: UUID.randomUUID().toString(),
+            "send",
+            Event.EventAction.NONE
```

Comment:
In reference to Qian's comments: This is what is causing the file to get uploaded to a "folder" called none, right?

Reply:
Yep, that's correct